### ISSUE TITLE
QT5 Error on Fedora 34 / Wayland

### ISSUE THREAD

**Issue by AstroStock** [3fon3fonov/exostriker#79]
Installing Exostriker on Fedora 34 under Wayland leads to the following problem when running exostriker:

> ImportError: /lib64/libQt5Core.so.5: version `Qt_5_PRIVATE_API' not found (required by /home/lukasstock/anaconda3/lib/python3.8/site-packages/PyQt5/QtCore.abi3.so)

Installation itself works fine.

QT5 is installed:

> locate libQt5Quick.so.5
> /usr/lib64/libQt5Quick.so.5
> /usr/lib64/libQt5Quick.so.5.15
> /usr/lib64/libQt5Quick.so.5.15.2 

Seems the problem is related to the QT Version, as I saw similar problems with QT with e.g. zoom

Running

> sudo pip3 uninstall PyQt5
> sudo pip3 install PyQt5==5.15.4

didn't solve the problem.

I'm running python under anaconda. Maybe this will help:

> conda list Qt
>
>  packages in environment at /home/XXX/anaconda3:
> 
>  Name                    Version                   Build  Channel
> pyqt                      5.9.2            py38h05f1152_4  
> pyqt5                     5.15.4                   pypi_0    pypi
> pyqt5-qt5                 5.15.2                   pypi_0    pypi
> pyqt5-sip                 12.9.0                   pypi_0    pypi
> qt                        5.9.7                h5867ecd_1  
> qtawesome                 1.0.2              pyhd3eb1b0_0  
> qtconsole                 5.1.0              pyhd3eb1b0_0  
> qtpy                      1.9.0                      py_0  
> sphinxcontrib-qthelp      1.0.3              pyhd3eb1b0_0  

and

> dnf list python3-qt*
>
> Installierte Pakete
> python3-qt5.x86_64                          5.15.0-8.fc34                @fedora
> python3-qt5-base.x86_64                     5.15.0-8.fc34                @fedora
> Verfügbare Pakete
> python3-QtAwesome.noarch                    1.0.2-2.fc34                 fedora 
> python3-QtPy.noarch                         1.9.0-6.fc34                 fedora 
> python3-qt5.i686                            5.15.0-8.fc34                fedora 
> python3-qt5-base.i686                       5.15.0-8.fc34                fedora 
> python3-qt5-devel.i686                      5.15.0-8.fc34                fedora 
> python3-qt5-devel.x86_64                    5.15.0-8.fc34                fedora 
> python3-qt5-webengine.x86_64                5.15.2-1.fc34                fedora 
> python3-qt5-webkit.x86_64                   5.15.0-8.fc34                fedora 
> python3-qtconsole.noarch                    5.0.3-1.fc34                 updates
> python3-qtsass.noarch                       0.1.1-4.fc34                 fedora 

Thank you very much for your help in advance

**Comment by 3fon3fonov** [comment]
I might have to install Fedora on a virtual machine and try to reproduce the problem and debug. I suspect you have system PyQt5, which likely must be removed to  use the one you install via pip.

First please tell me how did you install the Exo-Striker? via pip or just "git clone" and then you do, e.g.,: 
$python3 exostriker_gui.py 

(see instructions)

anyway, please do 

$exostriker -debug

or in the root directory of your git clone

$ python3 exostriker_gui.py -debug

and report the output.

**Comment by AstroStock** [comment]
I installed it via:

> git clone https://github.com/3fon3fonov/exostriker   
> cd exostriker
> python3 setup.py install 

Running exostriker  - debug:

> Traceback (most recent call last):
>   File "/home/lukasstock/anaconda3/bin/exostriker", line 4, in <module>
>     __import__('pkg_resources').run_script('exostriker==0.61', 'exostriker')
>   File "/home/lukasstock/anaconda3/lib/python3.8/site-packages/pkg_resources/__init__.py", line 651, in run_script
>     self.require(requires)[0].run_script(script_name, ns)
>   File "/home/lukasstock/anaconda3/lib/python3.8/site-packages/pkg_resources/__init__.py", line 1448, in run_script
>     exec(code, namespace, namespace)
>   File "/home/lukasstock/anaconda3/lib/python3.8/site-packages/exostriker-0.61-py3.8.egg/EGG-INFO/scripts/exostriker", line 2, in <module>
>     import exostriker
>   File "/home/lukasstock/anaconda3/lib/python3.8/site-packages/exostriker-0.61-py3.8.egg/exostriker/__init__.py", line 1, in <module>
>     from exostriker import gui
>   File "/home/lukasstock/anaconda3/lib/python3.8/site-packages/exostriker-0.61-py3.8.egg/exostriker/gui.py", line 13, in <module>
>     from PyQt5 import QtCore, QtGui, QtWidgets, uic
> ImportError: /lib64/libQt5Core.so.5: version `Qt_5_PRIVATE_API' not found (required by /home/lukasstock/anaconda3/lib/python3.8/site-packages/PyQt5/QtCore.abi3.so)

and python3 exostriker_gui.py -debug:

>  File "exostriker_gui.py", line 2, in <module>
>     import exostriker
>   File "/home/lukasstock/Dokumente/Masterarbeit/exostriker/exostriker/__init__.py", line 1, in <module>
>     from exostriker import gui
>   File "/home/lukasstock/Dokumente/Masterarbeit/exostriker/exostriker/gui.py", line 13, in <module>
>     from PyQt5 import QtCore, QtGui, QtWidgets, uic
> ImportError: /lib64/libQt5Core.so.5: version `Qt_5_PRIVATE_API' not found (required by /home/lukasstock/anaconda3/lib/python3.8/site-packages/PyQt5/QtCore.abi3.so)

**Comment by 3fon3fonov** [comment]
Ok, this is a deep problem not related to the Exo-Striker. It hangs very early in the code and is not even able to print the dependencies list, which I want you to inspect.

try:

$ QT_SELECT=5 qmake -v

mine is :

trifonov@lnx:~$ QT_SELECT=5 qmake -v
QMake version 3.1
Using Qt version 5.15.0 in /usr/local/lib/python3.8/dist-packages/pyqt5_tools/Qt/lib

if you see anything other than what you installed with pip (i.e. PyQt5==5.15.4), then you should remove it. 

in Fedora it should be something like":

$ dnf remove PyQt5

but I am not really familiar with any of the Red Hat distributions. 

Generally, I recommend Debian/Ubuntu distributions to avoid any complicated issues that may arise.

**Comment by AstroStock** [comment]
Well, that's interesting:

> QT_SELECT=5 qmake -v
> 
> QMake version 3.1
> Using Qt version 5.9.7 in /home/lukasstock/anaconda3/lib

Version 5.15 doesn't show up at all

**Comment by 3fon3fonov** [comment]
So remove all PyQt5 versions by e.g.:

$ sudo pip3 uninstall PyQt5 

you must make sure pip3 is the one tied to anaconda3 and repeat until no PyQt5 version is left

then:

$ sudo pip3 install PyQt5 ==5.15.4 (or 5.15.0, the one that seems to work on my Linux distro with no problem)


Please report your progress.

**Comment by AstroStock** [comment]
After

> sudo pip3 uninstall PyQt5 
the problem still stays the same and QT_SELECT=5 qmake -v also doesn't change

After running

> conda remove pt

it doen't find qmake anymore.

After than installing 

> sudo pip3 install PyQt5==5.15.0

qmake isn't found either. I think is is therefore a problem between conda and pip3

running exostriker -debug now raises:

> Traceback (most recent call last):
>   File "/home/lukasstock/anaconda3/bin/exostriker", line 4, in <module>
>     __import__('pkg_resources').run_script('exostriker==0.61', 'exostriker')
>   File "/home/lukasstock/anaconda3/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3243, in <module>
>     def _initialize_master_working_set():
>   File "/home/lukasstock/anaconda3/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3226, in _call_aside
>     f(*args, **kwargs)
>   File "/home/lukasstock/anaconda3/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3255, in _initialize_master_working_set
>     working_set = WorkingSet._build_master()
>   File "/home/lukasstock/anaconda3/lib/python3.8/site-packages/pkg_resources/__init__.py", line 568, in _build_master
>     ws.require(__requires__)
>   File "/home/lukasstock/anaconda3/lib/python3.8/site-packages/pkg_resources/__init__.py", line 886, in require
>     needed = self.resolve(parse_requirements(requirements))
>   File "/home/lukasstock/anaconda3/lib/python3.8/site-packages/pkg_resources/__init__.py", line 772, in resolve
>     raise DistributionNotFound(req, requirers)
> pkg_resources.DistributionNotFound: The 'qtconsole' distribution was not found and is required by exostriker

**Comment by AstroStock** [comment]
I installed all the packages EXOstriker wanted via conda and am now again at

> ImportError: /lib64/libQt5Core.so.5: version `Qt_5_PRIVATE_API' not found (required by /home/lukasstock/anaconda3/lib/python3.8/site-packages/PyQt5/QtCore.abi3.so)

Other outputs:
> sudo dnf remove python3-qt5
> Keine Übereinstimmung für Argumente: python3-qt5
> Keine Pakete zum Entfernen markiert.
> Abhängigkeiten sind aufgelöst.
> Nichts zu tun.
> Fertig.

> 
> sudo pip3 uninstall PyQt5
> WARNING: Skipping PyQt5 as it is not installed.

> QT_SELECT=5 qmake -v
> QMake version 3.1
> Using Qt version 5.9.7 in /home/lukasstock/anaconda3/lib
> 

> conda list qt
>  packages in environment at /home/lukasstock/anaconda3:
> 
>  Name                    Version                   Build  Channel
> pyqt                      5.9.2            py38h05f1152_4  
> pyqt5                     5.15.4                   pypi_0    pypi
> pyqt5-qt5                 5.15.2                   pypi_0    pypi
> pyqt5-sip                 12.9.0                   pypi_0    pypi
> qt                        5.9.7                h5867ecd_1  
> qtconsole                 5.1.0              pyhd3eb1b0_0  
> qtpy                      1.9.0                      py_0  
>

**Comment by 3fon3fonov** [comment]
Frankly, I don't know. You need to update your qt/pyqt somehow. I tried to install Fedora these days but I have some other, VirtualBox-related problems and I was unable to test. I will have a look next week when I am back from vacation.

**Comment by 3fon3fonov** [comment]
Can't you just bypass anaconda3? Why don't you make a system python install, which I am convinced is far more stable? If you are the owner of this system (i.e. it is not owned by the LSW), I would encourage you to forget about anaconda and install python packages via system pip.  remove anaconda and install the tool via:

$ pip3 install git+https://github.com/3fon3fonov/exostriker 

which should install all the needed dependencies in one go (well, I hope :) )

**Comment by AstroStock** [comment]
I use anaconda simply because so I'm automatically running in a python environment which is NOT the system environment. Else I would have destroyed my whole system several times :)

Anyway, I solved it last night. I created a new conda environment via 

> conda activate

and then installed all needed packages via pip3. If I use this new environment ExoStriker is running!

Thank you very much for your help and have a nice vacation.
